### ISSUE TITLE
Strange connection issue on big sur 11.3 clean install

### ISSUE THREAD

**Issue by rbernardes** [objective-see/LuLu#355]
Im made a clean install of my big sur 11.3
After install lulu, all connections were extremely slow.
On macos console, without the firewall enabled, and internet working without issues:

> error	14:35:06.532919-0300	kernel	Sandbox: ContextStoreAgen(443) deny(1) mach-lookup com.apple.ocspd
> error	14:35:08.174946-0300	kernel	Sandbox: routined(396) deny(1) mach-lookup com.apple.Maps.MapsSync.store
> error	14:35:08.175539-0300	routined	error: XPC: synchronousRemoteObjectProxyWithErrorHandler encountered error: Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service on pid 0 named com.apple.Maps.MapsSync.store was invalidated." UserInfo={NSDebugDescription=The connection to service on pid 0 named com.apple.Maps.MapsSync.store was invalidated.}
> error	14:35:08.175849-0300	routined	error: XPC: synchronousRemoteObjectProxyWithErrorHandler encountered error: Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service on pid 0 named com.apple.Maps.MapsSync.store was invalidated." UserInfo={NSDebugDescription=The connection to service on pid 0 named com.apple.Maps.MapsSync.store was invalidated.}
> error	14:35:09.199074-0300	kernel	Sandbox: com.apple.WebKit(801) deny(1) mach-lookup com.apple.diagnosticd
> error	14:35:11.263911-0300	routined	error: XPC: synchronousRemoteObjectProxyWithErrorHandler encountered error: Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service on pid 0 named com.apple.Maps.MapsSync.store was invalidated." UserInfo={NSDebugDescription=The connection to service on pid 0 named com.apple.Maps.MapsSync.store was invalidated.}
> error	14:35:11.264565-0300	kernel	Sandbox: routined(396) deny(1) mach-lookup com.apple.Maps.MapsSync.store
> error	14:35:14.365097-0300	routined	error: XPC: synchronousRemoteObjectProxyWithErrorHandler encountered error: Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service on pid 0 named com.apple.Maps.MapsSync.store was invalidated." UserInfo={NSDebugDescription=The connection to service on pid 0 named com.apple.Maps.MapsSync.store was invalidated.}
> error	14:35:15.381313-0300	cfprefsd	rejecting read of { kCFPreferencesAnyApplication, rbernardes, kCFPreferencesAnyHost, /Users/rbernardes/Library/Preferences/.GlobalPreferences.plist, managed: 0 } from process 634 (Brave Browser Helper (GPU)) because accessing preferences outside an application's container requires user-preference-read or file-read-data sandbox access
> error	14:35:17.464961-0300	routined	error: XPC: synchronousRemoteObjectProxyWithErrorHandler encountered error: Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service on pid 0 named com.apple.Maps.MapsSync.store was invalidated." UserInfo={NSDebugDescription=The connection to service on pid 0 named com.apple.Maps.MapsSync.store was invalidated.}
> error	14:35:20.564407-0300	routined	error: XPC: synchronousRemoteObjectProxyWithErrorHandler encountered error: Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service on pid 0 named com.apple.Maps.MapsSync.store was invalidated." UserInfo={NSDebugDescription=The connection to service on pid 0 named com.apple.Maps.MapsSync.store was invalidated.}
> error	14:35:21.413165-0300	cfprefsd	rejecting read of { kCFPreferencesAnyApplication, rbernardes, kCFPreferencesAnyHost, /Users/rbernardes/Library/Preferences/.GlobalPreferences.plist, managed: 0 } from process 634 (Brave Browser Helper (GPU)) because accessing preferences outside an application's container requires user-preference-read or file-read-data sandbox access
> error	14:35:23.663295-0300	routined	error: XPC: synchronousRemoteObjectProxyWithErrorHandler encountered error: Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service on pid 0 named com.apple.Maps.MapsSync.store was invalidated." UserInfo={NSDebugDescription=The connection to service on pid 0 named com.apple.Maps.MapsSync.store was invalidated.}

When enabled:

> error	14:35:39.904803-0300	mDNSResponder	[Q0] mDNSPlatformSendUDP -> sendto(6) failed to send packet on InterfaceID 0x6 en0/4 to <mask.hash: 'kIpuBtg61p0TRx/X6fp11A=='>:5353 skt 6 error -1 errno 32 (Broken pipe) 3623211914
> error	14:35:40.764903-0300	kernel	Sandbox: com.apple.WebKit(790) deny(1) mach-lookup com.apple.diagnosticd
> error	14:35:46.729562-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:46.742723-0300	kernel	Sandbox: ContextStoreAgen(443) deny(1) mach-lookup com.apple.ocspd
> error	14:35:50.729568-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:50.970484-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:50.970604-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:50.970726-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:50.970833-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:50.971251-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:50.972994-0300	symptomsd	failed to start ticker tickerFatal, not active, active probe: 0x0
> error	14:35:50.973507-0300	symptomsd	failed to start ticker tickerFatal, not active, active probe: 0x0
> error	14:35:50.976296-0300	symptomsd	failed to start ticker tickerFatal, not active, active probe: 0x0
> error	14:35:50.976722-0300	symptomsd	failed to start ticker tickerFatal, not active, active probe: 0x0
> error	14:35:51.372396-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:51.373310-0300	symptomsd	failed to start ticker tickerFatal, not active, active probe: 0x0
> error	14:35:52.079577-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:52.080893-0300	symptomsd	failed to start ticker tickerFatal, not active, active probe: 0x0
> error	14:35:54.890142-0300	kernel	Sandbox: com.apple.WebKit(757) deny(1) mach-lookup com.apple.diagnosticd
> error	14:35:55.001122-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:55.001312-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:55.001404-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:55.001502-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:55.001589-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:55.503511-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:56.279367-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:59.001058-0300	mDNSResponder	sending to <IPv4:BBkywCMi> failed: [32: Broken pipe]
> error	14:35:59.003915-0300	symptomsd	failed to start ticker tickerFatal, not active, active probe: 0x0

Anyone with this problem too?

**Comment by tk176** [comment]
Yes, I found it out at this moment, that LuLu made the problems. Also other tool won't work. (for example macupdater or twitteriffice)

**Comment by SwiftWinds** [comment]
Also facing the same issue; something to note: LAN connections (i.e., to `192.168.x.y`) were fast, but WAN ones were unbearably slow.

**Comment by SwiftWinds** [comment]
FWIW, a reinstall did not work

**Comment by danielp123** [comment]
Having the same problem. WAN connections are unbearably slow. Ping does not work on WAN or LAN for that matter. I use an ethernet connection if that's important.

**Comment by clicman** [comment]
Same here.

**Comment by necrevistonnezr** [comment]
Same here.

**Comment by danielp123** [comment]
Solution that fixed it for me: remove Radio Silence. I guess that there are conflicts between different filtering apps that are appearing in Big Sur 11.3.
Check if you have other network filtering apps active and remove them.

**Comment by jdswinbank** [comment]
Seems like there are a lot of reports of LuLu having problems when used in combination with some other networking app. For example:

- Radio Silence (@danielp123, above)
- TripMode (https://github.com/objective-see/LuLu/issues/321#issuecomment-827159116, and my personal experience)
- Cisco AnyConnect (#356)

At least for me, _just_ running LuLu on macOS 11.3 seems to work fine. I wonder if any of the folks reporting problems are using LuLu on its own?

**Comment by clicman** [comment]
Eset Cyber Security Pro and AdGuard maybe also affects it.

**Comment by rbernardes** [comment]
+1 radio silence
They release an update, but not solved the problem

**Comment by SwiftWinds** [comment]
> Solution that fixed it for me: remove Radio Silence. I guess that there are conflicts between different filtering apps that are appearing in Big Sur 11.3.
> Check if you have other network filtering apps active and remove them.

Thanks for the suggestion. I was using Pulse Secure VPN. Removing it seems to do the trick, but I kind of need the app. :/

**Comment by inspiredearth** [comment]
Also having what I suspect is this same issue.

Running Lulu 2.3.1, on MacOS 11.3, on M1 chip. I also have TripMode running.

If Lulu is enabled, when I connect to WiFi it either comes up with `!` symbol over WiFi icon, or it does connect but then Internet is basically inaccessible. I'm pretty sure it's due to DNS lookups being blocked. As I note that I can load a common site in browser (such as Google.com) for which I know there will be an entry in the DNS cache.  And I can call other sites for which there is very likely an IP entry in the DNS cache. But once anything beyond what's cached is called, it doesn't load, or it takes many minutes to load.

Also using a VPN is problematic. I'd not associated it with this Lulu issue, but seeing comments above made me realise it likely is related.

**Comment by rbernardes** [comment]
Today apple release big sur 11.3.1.
Maybe this problem has solved, not testing yet.

**Comment by emory** [comment]
I haven't updated yet but I am not opposed to it. I just wanted to mention first that my employer-provided workstation is running Eracent for DLP I believe, and Crowdstrike as it's endpoint protection software and LuLu must be disabled or I can't ping anything and only certain activities are even possible. 

But I've got a trashcan6,1 on my desk too and it's fine with LuLu running. My endpoint protection software on that workstation is BitDefender and like I do ad-hoc vt-cli and clamscans but nothing in-path for networking other than BitDefender and LuLu is happy with that arrangement. 

I don't have much visibility into Eracent other than what I can figure out from it's logs, and nobody from our SIRT has contacted me about Crowdstrike melting down or anything like that so I think whatever the problem is I'm expecting the only solution for me will be a change in LuLu. I'm curious what the root cause here is though! I am def not up to speed on what changes under the hood have been made in macOS over the last couple of releases.

**Comment by clicman** [comment]
FYI 11.3.1 doesn't solve the problem.

**Comment by dguttierrez** [comment]
I also have problems with Big Sur 11.3.1 latest Lulu and backup to Time Machine on NAS.

**Comment by hasanpour** [comment]
Same ping issue here on Big Sur 11.3.1, Lulu 2.3.1 and Cisco AnyConnect 4.10.00093. I think this problem is related to new versions of Cisco AnyConnect after it starts using 2 Socket Filter services in network preferences.
<img width="624" alt="Screen Shot 2021-05-09 at 10 07 39 PM" src="https://user-images.githubusercontent.com/2172412/117581629-1b924d80-b113-11eb-8dc8-d5b671f6169c.png">

**Comment by dguttierrez** [comment]
I have Cisco AnyConnect 4.9.06037.

**Comment by winteriscariot** [comment]
I just confirmed by disconnecting the Cisco AnyConnect Packet Filter i have regained connectivity with LuLu activated. 

What do those packet filters even do? I was able to remove them and still connect to my AnyConnect VPN

**Comment by darkolo** [comment]
> Same ping issue here on Big Sur 11.3.1, Lulu 2.3.1 and Cisco AnyConnect 4.10.00093. I think this problem is related to new versions of Cisco AnyConnect after it starts using 2 Socket Filter services in network preferences.
> <img alt="Screen Shot 2021-05-09 at 10 07 39 PM" width="624" src="https://user-images.githubusercontent.com/2172412/117581629-1b924d80-b113-11eb-8dc8-d5b671f6169c.png">

Socket filtering is present in Cisco AnyConnect since version 4.9

**Comment by inspiredearth** [comment]
I don't have Cisco AnyConnect installed. So I suspect that's not the specific cause.

**Comment by MagicPhantom** [comment]
If enabled, LuLu won't do Backups in TimeMachine. If I disable it, Backups work without problems.

**Comment by inspiredearth** [comment]
On Big Sur, I’ve had to essentially keep Lulu disabled. I consider it completely incompatible with Big Sur. It causes far to many connection issues to be useable.

**Comment by comatose-tortoise** [comment]
Yes, same here. I have it disabled on Big Sur 11.3, otherwise nothing works.

**Comment by dtidmore52** [comment]
I am running macOS 11.x on a mid 2012 15" non retina MBP installed with patch-sur utility.  I had upgraded my WiFi/BT card to the newest Broadcom version (same as apple is currently using). Versions of macOS 11.2.5 and before ran perfectly but all hell broke loose when I upgraded to 11.3.  I chased it to the point of determining it was definitely related to DNS.  I then fell back to macOS11.2.5 successfully and the issue was resolved.  I chose at the time to just stay on 11.2.5 and try again with 11.4.  So, today Apple released 11.4 and I upgraded.  Same painfully slow DNS resolution once again raised its ugly head.  Searching the internet once again (on my iPhone), resulted in finding a reddit thread from a M1 based MBP user reporting the same problem which meant, the issue was most decidedly not related to my hardware.  That user had chased the issue to LuLu, which I have used for a couple of years without issues.  I removed LuLu (ver 2.3.1), rebooted and BINGO, DNS resolution returned to normal expectations.  What LuLu is doing to so screw up DNS resolution starting with macOS 11.3 is something Patrick W will have to sort out.  For the meantime, I will just keep LuLu off my system but I DO hope Patrick finds the issue and resolves it as I prefer to run LuLu.

**Comment by darkolo** [comment]
Same issue with 11.4

**Comment by BartmanEH** [comment]
> Same issue with 11.4

bummer. for now I've disabled the three packet filters network interfaces for Cisco AnyConnect v4.9.xxxx and that seems to allow Cisco Connect to still work as a vpn client as well as allowing LuLu to work as a firewall and restoring ping functionality etc.

**Comment by darkolo** [comment]
After updating Big Sur to 11.5 it seems that Lulu is working again and I can ping

**Comment by clicman** [comment]
Confirming. It works on 11.5!

<img width="837" alt="image" src="https://user-images.githubusercontent.com/1079535/126740921-7232ba99-9330-4269-945e-00d707fbc3b1.png">

**Comment by inspiredearth** [comment]
I can confirm the same ... Issue seems to be resolved on 11.5 (and 11.5.2)

**Comment by BartmanEH** [comment]
Interestingly [release notes for new AnyConnect client](https://www.cisco.com/c/en/us/td/docs/security/vpn_client/anyconnect/anyconnect410/release/notes/release-notes-anyconnect-4-10.html#Cisco_Reference.dita_41127f2b-432f-4cc8-8978-c12a9f7db890) state:
AnyConnect 4.10.05095 New Features
This is a maintenance release that includes the following enhancements, and that resolves the defects described in [AnyConnect 4.10.05095](https://www.cisco.com/c/en/us/td/docs/security/vpn_client/anyconnect/anyconnect410/release/notes/release-notes-anyconnect-4-10.html#Cisco_Reference.dita_88f7d339-2a94-4119-b221-8ac30815bea3).
[...redacted, irrelevant...]
_**An Umbrella issue that could cause a total Domain Name System failure on macOS 11 and later versions, requiring a reboot or removal of AnyConnect to resolve, has been fixed.**_
...so maybe we can re-enable the Cisco AnyConnect packet filters (although I probably won't bother since the VPN part works fine without the packet filters enabled)

**Comment by Anutrix** [comment]
Same issue along with https://www.paloaltonetworks.com/products/globalprotect. Not sure if 11.5 fixed it.
